src/serverMock.ts

```diff
@@ -0,0 +1,82 @@
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+import { gql } from '@apollo/client';
+import { buildClientSchema, execute } from 'graphql';
+import { addMocksToSchema } from '@graphql-tools/mock';
+import introspection from './api/introspection.json';
+
+import { config } from './config';
+import {
+  ProspectType,
+  ScheduledCorpusItem,
+  ScheduledSurface,
+} from './api/generatedTypes';
+
+// Build a schema using the introspection
+
+// The below is needed for as long as this issue remains open:
+// https://github.com/graphql/graphql-js/issues/3409
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+const schema = buildClientSchema(introspection);
+
+// Stub out our schema with fake data
+const mockedSchema = addMocksToSchema({
+  schema,
+  mocks: {
+    // The good thing here is that there is no need to mock the query
+    // AND we can type the query result with types from generatedTypes.ts
+    // The not so good thing is that the responses still need to be fully mocked,
+    // even in this one central place!
+    // TODO: add all available surfaces here
+    getScheduledSurfacesForUser: (): ScheduledSurface[] => [
+      {
+        name: 'New Tab (en-US)',
+        guid: 'NEW_TAB_EN_US',
+        ianaTimezone: 'America/New_York',
+        prospectTypes: [
+          ProspectType.Global,
+          ProspectType.OrganicTimespent,
+          ProspectType.SyndicatedNew,
+          ProspectType.SyndicatedRerun,
+          ProspectType.CountsLogisticApproval,
+          ProspectType.HybridLogisticApproval,
+          ProspectType.Approved,
+          ProspectType.TimespentLogisticApproval,
+        ],
+      },
+      {
+        name: 'New Tab (de-DE)',
+        guid: 'NEW_TAB_DE_DE',
+        ianaTimezone: 'Europe/Berlin',
+        prospectTypes: [
+          ProspectType.Global,
+          ProspectType.OrganicTimespent,
+          ProspectType.DomainAllowlist,
+        ],
+      },
+    ],
+    // TODO: to be mocked in more detail (:
+    getScheduledCorpusItems: (): ScheduledCorpusItem[] => [],
+  },
+});
+
+// Set up a server that reads a GraphQL document and returns the data for it
+setupServer(
+  rest.post<{ query: string; variables: any }>(
+    config.adminApiEndpoint,
+    async (req, res, ctx) => {
+      const result = await execute(
+        mockedSchema,
+        gql`
+          ${req.body.query}
+        `,
+        null,
+        null,
+        req.body.variables
+      );
+
+      return res(ctx.json(result));
+    }
+  )
+);
```

Comment:
@kschelonka, I'm open to this approach but I think the idea behind writing out mocks is that we'd be able to have different mocks (for example, one scheduled surface available for a user rather than all of them) for different integration tests, allowing for more flexibility. 

This pilot integration test suite puts all the mocks in a central location but we've discussed it with @jpetto earlier today and agreed it might be better to put just the mocks we need alongside/inside the integration tests instead.

Reply:
Yeah I was mostly thinking about possibly reducing the labor required by mocking fields you don't care about as much automatically (just for your types basically), and doing overrides for specific test cases